cmd/api/app/routes/member/deployment/handler.go

```diff
@@ -0,0 +1,76 @@
+package deployment
+
+import (
+	"context"
+
+	"github.com/gin-gonic/gin"
+	"github.com/karmada-io/dashboard/cmd/api/app/router"
+	"github.com/karmada-io/dashboard/cmd/api/app/types/common"
+	"github.com/karmada-io/dashboard/pkg/client"
+	"github.com/karmada-io/dashboard/pkg/resource/deployment"
+	"github.com/karmada-io/dashboard/pkg/resource/event"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func handleGetMemberDeployments(c *gin.Context) {
+	karmadaClient := client.InClusterKarmadaClient()
+	_, err := karmadaClient.ClusterV1alpha1().Clusters().Get(context.TODO(), c.Param("clustername"), metav1.GetOptions{})
+	if err != nil {
+		common.Fail(c, err)
+		return
+	}
```

Comment:
duplicate check for existence of member cluster, for these kind of check, we can use middleware in gin, like :
```
// middleware.go
func EnsureMemberClusterMiddleware() gin.HandlerFunc {
	return func(c *gin.Context) {
		karmadaClient := client.InClusterKarmadaClient()
		_, err := karmadaClient.ClusterV1alpha1().Clusters().Get(context.TODO(), c.Param("clustername"), metav1.GetOptions{})
		if err != nil {
			c.AbortWithStatusJSON(http.StatusOK, common.BaseResponse{
				Code: 500,
				Msg:  err.Error(),
			})
			return
		}
		c.Next()
	}
}

// usage in setup.go
member = v1.Group("/member/:clustername")
member.Use(EnsureMemberClusterMiddleware())
```
when the code executing in the handler, we can ensure the existence of membercluster, otherwise it will be abort in the middleware.